src/apis/impl/CodingApiImpl.ts

```diff
@@ -8,50 +8,68 @@ import {CodingMapper} from "../../mappers/codeCoding";
 import {PaginatedListMapper} from "../../mappers/paginatedList";
 import {FilterMapper} from "../../mappers/filter";
 import {firstOrNull} from "../../utils/functionalUtils";
+import {ErrorHandler} from "../../services/ErrorHandler";
 
 export class CodingApiImpl implements CodingApi {
   private readonly codeApi: IccCodeApi;
+  private readonly errorHandler: ErrorHandler;
 
-  constructor(api: { codeApi: IccCodeXApi}) {
+  constructor(api: { codeApi: IccCodeXApi }, errorHandler: ErrorHandler) {
     this.codeApi = api.codeApi
+    this.errorHandler = errorHandler;
   }
 
   async createOrModifyCoding(coding: Coding): Promise<Coding> {
-    const processedCoding = firstOrNull(await this.createOrModifyCodings([coding]));
+    const processedCoding = firstOrNull(await this.createOrModifyCodings([coding]).catch(e => {
+      throw this.errorHandler.createErrorFromAny(e)
+    }));
     if (processedCoding != undefined) {
       return processedCoding;
     } else {
-      throw new Error("Couldn't create or update coding");
+      throw this.errorHandler.createErrorWithMessage("Couldn't create or update coding");
```

Comment:
For the rest of the messages, take example of what I created in the AuthenticationApiImpl : You need to focus on what the user needs to do to resolve this error. He knows that he could not create oe update codings, but what does he needs to do to create / update them then ? 
It can even be a simple retry, but let's give them a solution, a next step in our error messages